src/plugin/index.js

```diff
@@ -56,12 +62,13 @@ Jarvis.prototype.apply = function(compiler) {
     plugin => plugin.constructor.name === "DefinePlugin"
   )[0];
 
-  const pluginNodeEnv = definePlugin["definitions"]["process.env.NODE_ENV"];
-
-  if (definePlugin && typeof pluginNodeEnv !== "undefined") {
-    pluginNodeEnv === "production"
-      ? (this.env.production = true)
-      : (this.env.production = false);
+  if (definePlugin) {
+    const pluginNodeEnv = definePlugin["definitions"]["process.env.NODE_ENV"];
```

Comment:
Hey @zouhir, my bad for breaking this. I meant to do this...

```
const definePlugin = compiler.options.plugins.filter(
    plugin => plugin.constructor.name === "DefinePlugin"
)[0];
```
```javascript
const pluginNodeEnv = `!definePlugin ? undefined : ["definitions"]["process.env.NODE_ENV"];`
```

```
if (definePlugin && typeof pluginNodeEnv !== "undefined") {
    pluginNodeEnv === "production"
      ? (this.env.production = true)
      : (this.env.production = false);
}
```

I've tested this and it works. Not trying to push my agenda, was just trying to improve readability. Either is fine.

Reply:
Hey @pickra ha! All good you didn't actually break anything!

I'm thinking that soon we will need a utility to get plugin data as I can see this dashboard integrating with more stuff real soon.

```
let thatPlugin = getplugin('ThatPluginName');
```

And we'll find a way to handle stuff outside the function body real nice!

For now, I'm keen to release a quick fix for this, and I'd keep it as is as it looks super decent to me!

If you have tested it, and happy with it for now, we can merge an open a ticket for handling plugins stuff in a more neat way?

Reply:
I think only checking defaultPlugin misses when the environment is set via cli and not w/definePlugin. If the objective is to check if the current build is production we could:
- Check cli arguments for `-p`, `process.env.NODE_ENV='production'`, `--env` value, etc...
- If cli arguments doesn't help, then check defaultPlugin
- Check if webpack emits something? Need to dig deeper to see if this is an option.

If the objective is to get plugin data, I could use more info.